.changes/unreleased/Under the Hood-20230119-103909.yaml

```
@@ -0,0 +1,9 @@
+kind: Under the Hood
+body: 'expand logic from #389 to try and make to config option tmp_relation_type as
+  optional as possible while maintining saftey of table as default and adding a few
+  exceptions to raise in rare casses with clear messages on fix.'
```

Can we make this more clear and succinct?

One suggestion
```suggestion
body: 'Enable a `tmp_relation_type` config for applicable incremental models'
```